astropy/table/ndarray_mixin.py

```diff
@@ -0,0 +1,126 @@
+        (for N > 1) by optionally turning it into the required number of 1-d
+        arrays (e.g. for ECSV or None context).
+
+        For instance a Column with shape (2, 2, 20) will turn into four columns
+        (each length 20) with "data attribute" names 0_0, 0_1, 1_0, and 1_1.
+        These then get constructed back into the same (2, 2, 20) Column.
+        """
+        col = self._parent
+
+        if col.ndim > 1 and self._serialize_context in FLATTEN_MULTIDIM:
+            # N-d column case for output to ECSV
+            out = {}
+            # Make cartesian product of all dimensions except length (first axis)
+            ranges = [range(ii) for ii in col.shape[1:]]
+            for dims in itertools.product(*ranges):
+                name = '_'.join(str(ii) for ii in dims)
+                out[name] = col[(...,) + dims].view(np.ndarray)
+        else:
+            out = {'data': col.view(np.ndarray)}
+
+        return out
+
+    def _construct_from_dict(self, map):
+        """Construct Column from ``map``.
+
+        Normally this is done by the default constuctor in DataInfo.
+
+        However as a special case if there are any keys in ``map`` that begin
+        with a digit, this must correspond to an N-d column that was serialized
+        through ECSV. (No actual attribute names can begin with a digit).
+        """
+        idxs_map = {}
+        # Look for keys like 0_0 or 2_1 corresponding to serializing an N-d
+        # Column. For instance a Column with shape (2, 2, 20) will turn into
+        # four columns (each length 20) with "data attribute" names 0_0, 0_1,
+        # 1_0, and 1_1.
+        for key in map:
+            if re.match(r'\d', key):
+                idxs = tuple(int(ii) for ii in key.split('_'))
+                idxs_map[key] = idxs
+
+        if idxs_map:
+            # Found appropriately-named keys, so this is an N-d column. Get the
+            # length of the first Column in the set of N-d columns (they must
+            # all be the same length).
+            key0 = next(iter(map))
+            data0 = map[key0]
+            length = len(data0)
+
+            # Get the shape by using finding max of the index tuples (tuples do
+            # indeed sort correctly).
+            max_idxs = max(idxs_map.values())
+            shape = (length,) + tuple(ii + 1 for ii in max_idxs)
+
+            # Make the N-d array and fill accordingly
+            data = np.empty(shape, dtype=data0.dtype)
+            for key, idxs in idxs_map.items():
+                data[(...,) + idxs] = map.pop(key)
+        else:
+            data = map.pop('data')
+
+        out = self._parent_cls(data, **map)
+
+        return out
+
+
+class NdarrayMixin(np.ndarray):
+    """
+    Mixin column class to allow storage of arbitrary numpy
+    ndarrays within a Table.  This is a subclass of numpy.ndarray
+    and has the same initialization options as ndarray().
+    """
+    __module__ = 'astropy.table.table'
```

Comment:
This class definition is the same as the original that was taken out of `table.py`, with the exception of this line and the one below. I don't remember precisely what hacking the `__module__` here was doing, but it might have been for the docs.